config/mapprovider.go

```diff
@@ -80,11 +80,19 @@ type ChangeEvent struct {
 }
 
 // Retrieved holds the result of a call to the Retrieve method of a Provider object.
-type Retrieved struct {
-	// Deprecated: [v0.50.0] Use NewRetrievedFromMap to initialize, and Retrieved.AsMap to access.
-	*Map
-	// Deprecated: [v0.50.0] Use NewRetrievedFromMap to initialize, and Retrieved.Close to access.
-	CloseFunc
+type Retrieved interface {
```

Comment:
Alternative, can make this a struct with only internal members for the moment.

Reply:
I don't have strong feelings either way... it feels like we should have some criteria on when unimplementable interface is preferable  vs. when do we prefer a struct with private fields but, lacking that, this is fine to me

Reply:
In general I think unless 2 completely different implementation of the same thing you should not use an interface. Let me try the other version to compare :)

Reply:
First trivial change is here https://github.com/open-telemetry/opentelemetry-collector/pull/5363, this should be non controversial. After that is merged I will make another PR to show how we can have with a struct the 2 possible values vs interface.

Reply:
I kind of implement the interface with only one struct, see this updated PR, I think we don't necessary need the "interface" change anymore. Waiting for feedback.